### ISSUE TITLE
PlugIn does not work when repository is inside WSL

### ISSUE THREAD

**Issue by Cutuchiqueno** [oxygenxml/oxygen-git-client-addon#154]
I am working on Windows 10 and 11, having my repositories inside *Windows Subsystem for Linux 2*. When I want to use the git plugin in such a project/repository, oxygen first shows a message that says that a repository was recognised and if I want to use the repository. After I confirmed nothing happens, no commit history, no changes tracked etc.

**Comment by chrispy-snps** [comment]
@Cutuchiqueno - is your repo in the Linux filesystem or the Windows filesystem?

**Comment by Cutuchiqueno** [comment]
yes, that's true.

**Comment by Cutuchiqueno** [comment]
no, sorry, that was definitely on me 😀. It's on thebLinux filesystemn(in WSL)

**Comment by chrispy-snps** [comment]
@Cutuchiqueno - great, thanks!

To narrow down the issue, could you create a copy of the repo directory in the Windows filesystem and see if the Git machinery works there? Maybe the WSL2 linux filesystem isn't providing the needed filesystem information for JGit to scan for changes, and this experiment would tell us if that's the case.

**Comment by Cutuchiqueno** [comment]
Ok, I took the opportunity to evaluate the case more comprehensively. As it turned out, it is severe performance issue.

I tried 2 repositories, a) with 14MB in size and b) with 2,7GB (the one that raised the issue), both in the WSL and in the Windows file system.

Both work more or less fine within Windows filesystem, though b) is a bit lacky. In WSL a) is lacky and b) never even gets to the point where it indexes even the different branches of the repo.
